src/migrations/20221010162913-add-self-trade-type.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint @typescript-eslint/ban-ts-comment: "off" */
+// @ts-nocheck
+
+/* eslint @typescript-eslint/no-var-requires: "off" */
+const { Types } = require("mongoose")
+
+const LedgerTransactionType = {
+  IntraLedger: "on_us",
+  LnIntraLedger: "ln_on_us",
+  OnchainIntraLedger: "onchain_on_us",
+  WalletIdTradeIntraAccount: "self_trade",
+  LnTradeIntraAccount: "ln_self_trade",
+  OnChainTradeIntraAccount: "onchain_self_trade",
+}
+
+const toWalletId = (walletIdPath) => {
+  const path = walletIdPath.split(":")
+  return Array.isArray(path) && path.length === 2 && path[0] === "Liabilities" && path[1]
+    ? path[1]
+    : undefined
+}
+
+const getNonUserWalletIds = async ({ usersCollection, walletsCollection }) => {
+  const accountIds = []
+  const usersCursor = usersCollection.find({
+    role: { $in: ["funder", "dealer", "bankowner"] },
+  })
+  for await (const user of usersCursor) {
+    const accountId = user._id.toString()
+    accountIds.push(accountId)
+  }
+  const accountObjectIds = accountIds.map((id) => new Types.ObjectId(id))
+
+  const nonUserWalletIds = []
+  const walletsCursor = walletsCollection.find({
+    _accountId: { $in: accountObjectIds },
+  })
+  for await (const wallet of walletsCursor) {
+    nonUserWalletIds.push(wallet.id)
+  }
+  return nonUserWalletIds
+}
+
+module.exports = {
+  async up(db) {
+    const usersCollection = db.collection("users")
+    const walletsCollection = db.collection("wallets")
+    const txnsCollection = db.collection("medici_transactions")
+
+    const nonUserWalletIds = await getNonUserWalletIds({
+      usersCollection,
+      walletsCollection,
+    })
+
+    const accountIdsByWalletId = {}
+    const getSelfTradeType = async (txnAgg) => {
+      // BUILD 'txns' FOR JOURNALID
+      // =====
+      const txns = []
+      const txnsCursor = txnsCollection.find({ _journal: txnAgg._id })
+      for await (const txn of txnsCursor) {
+        // Filter internal wallet IDs
+        if (nonUserWalletIds.includes(toWalletId(txn.accounts))) continue
```

Comment:
not a blocker but you can also use `account_path` (it is an array so you dont have to split)